libs/libskynet/src/checkObj.ts

```diff
@@ -0,0 +1,35 @@
+import { Err } from "./types.js";
+
+// checkObj take an untrusted object and a list of typechecks to perform and
+// will check that the object adheres to the typechecks. If a type is missing
+// or has the wrong type, an error will be returned. This is intended to be
+// used to check untrusted objects after they get decoded from JSON. This is
+// particularly useful when receiving objects from untrusted entities over the
+// network or over postMessage.
+//
+// Below is an example object, followed by the call that you would make to
+// checkObj to verify the object.
+//
+// const expectedObj = {
+//   aNum: 35,
+//   aStr: "hi",
+//   aBig: 10n,
+// };
+//
+// const err = checkObj(expectedObj, [
+//   ["aNum", "number"],
+//   ["aStr", "string"],
+//   ["aBig", "bigint"],
+// ]);
+function checkObj(obj: any, checks: [string, string][]): Err {
```

Comment:
I think that's too verbose

Reply:
agreed, checkObj is just too generic, it's better to be verbose than generic, especially when this is exported function

Reply:
I'd go for `validatePropTypes`, though -- easier to read and verbose enough, imho :)

Reply:
that does close off any thought of extending the function later to support more types of checks on the object. For example, it could support verifying a checksum that's expected to be contained within the object. We already have plans to extend it so it can check for things like arrays, which is beyond the scope of `typeof` - I do think a generic name makes sense here.

The full idea behind this function is that we're going to be verifying an untrusted object, checking to make sure that it adheres to constraints that get provided as input.

Reply:
I also don't like calling it `validate` because that makes it seem like the Object is fully validated. But really it's only checking the things you tell it to check, we don't provide a clean guarantee that the object doesn't have other properties or manipulations

Reply:
I ended up calling it `reviewObjectPropTypes`

Reply:
I ended up calling it `validateUntrustedObject`

Reply:
changed to `validateObjPropTypes`